kolibri/core/assets/src/views/CoreBase/index.vue

```diff
@@ -333,16 +334,54 @@
         return '';
       },
     },
+    watch: {
+      startingScroll(newVal) {
+        // Set a watcher so that if the router sets a new
+        // starting scroll position based on the history, then it gets
+        // set here.
+        if (this.unwatchScrollHeight) {
+          this.unwatchScrollHeight();
+        }
+        if (this.loading) {
+          // Don't set scroll position until the main content
+          // of coreBase is shown in the DOM.
+          // Create a watcher to monitor changes in loading
+          // to try to set the scrollHeight after the contents
+          // have loaded.
+          this.unwatchScrollHeight = this.$watch('loading', () => {
```

Comment:
I wonder if `unwatchScrollHeight` should be defined on `data` and initialized to `undefined` for explicitness? Seems a little weird to just slap it onto the view model.

Reply:
Why? It doesn't need to be reactive, putting it on the view itself is exactly the right place for it, the only thing that might be better is prefixing it with an underscore, but I really don't see any benefit to having it be a watched bit of internal state.

Reply:
Yeah agreed – I don't like the fact that it's reactive either. The thing that bothered me is to add it to the `vm` namespace without declaring it anywhere.

There's a [lively discussion here](https://github.com/vuejs/vue/issues/1988) so apparently this is something that keeps people up at night :)

Looks like the original strategy you used is likely fine, and another common pattern is to put it inside `$options`

Reply:
I do find it a little odd that the github issue is so long. Either you put it in a namespace, or you put it right on the object. Vue doesn't, and shouldn't care!